tutorials/multi/2_node_classification.py

```diff
@@ -0,0 +1,259 @@
+        self.conv2 = SAGEConv(h_feats, num_classes, aggregator_type='mean')
+        self.h_feats = h_feats
+
+    def forward(self, mfgs, x):
+        h_dst = x[:mfgs[0].num_dst_nodes()]
+        h = self.conv1(mfgs[0], (x, h_dst))
+        h = F.relu(h)
+        h_dst = h[:mfgs[1].num_dst_nodes()]
+        h = self.conv2(mfgs[1], (h, h_dst))
+        return h
+
+
+######################################################################
+# Defining Training Procedure
+# ---------------------------
+# 
+# The training procedure will be slightly different from what you saw
+# previously, in the sense that you will need to
+#
+# * Initialize a distributed training context with ``torch.distributed``.
+# * Wrap your model with ``torch.nn.parallel.DistributedDataParallel``.
+# * Add a ``use_ddp=True`` argument to the DGL dataloader you wish to run
+#   together with DDP.
+# 
+# You will also need to wrap the training loop inside a function so that
+# you can spawn subprocesses to run it.
+# 
+
+def run(proc_id, devices):
+    # Initialize distributed training context.
+    dev_id = devices[proc_id]
+    dist_init_method = 'tcp://{master_ip}:{master_port}'.format(master_ip='127.0.0.1', master_port='12345')
+    # To use Multi-GPU, change 'gloo' to 'nccl'.
+    torch.distributed.init_process_group(
+        backend='gloo', init_method=dist_init_method, world_size=len(devices), rank=proc_id)
+    device = torch.device('cpu')
+    # Uncomment the following two lines to use GPU.
+    #torch.cuda.set_device(dev_id)
+    #device = torch.device('cuda:' + str(dev_id))
+    
+    # Define training and validation dataloader, copied from the previous tutorial
+    # but with one line of difference: use_ddp to enable distributed data parallel
+    # data loading.
+    sampler = dgl.dataloading.MultiLayerNeighborSampler([4, 4])
+    train_dataloader = dgl.dataloading.NodeDataLoader(
+        # The following arguments are specific to NodeDataLoader.
+        graph,              # The graph
+        train_nids,         # The node IDs to iterate over in minibatches
+        sampler,            # The neighbor sampler
+        device=device,      # Put the sampled MFGs on CPU or GPU
+        use_ddp=True,       # Make it work with distributed data parallel
+        # The following arguments are inherited from PyTorch DataLoader.
+        batch_size=1024,    # Per-device batch size.
+                            # The effective batch size is this number times the number of GPUs.
+        shuffle=True,       # Whether to shuffle the nodes for every epoch
+        drop_last=False,    # Whether to drop the last incomplete batch
+        num_workers=0       # Number of sampler processes
+    )
+    valid_dataloader = dgl.dataloading.NodeDataLoader(
+        graph, valid_nids, sampler,
+        batch_size=1024,
+        shuffle=False,
+        drop_last=False,
+        num_workers=0,
+        device=device,
+        use_ddp=False
+    )
```

Comment:
It's better to use a consistent order of the arguments for both `train_dataloader` and `valid_dataloader`.